controllers/discordactions.js

```diff
@@ -67,25 +66,19 @@ const createGroupRole = async (req, res) => {
 const getAllGroupRoles = async (req, res) => {
   try {
     const { groups } = await discordRolesModel.getAllGroupRoles();
-    const groupsWithMemberCount = await discordRolesModel.getNumberOfMemberForGroups(groups);
-    const groupCreatorIds = groupsWithMemberCount.reduce((ids, group) => {
-      ids.add(group.createdBy);
-      return ids;
-    }, new Set());
-    const groupCreatorsDetails = await retrieveUsers({ userIds: Array.from(groupCreatorIds) });
-    const groupsWithUserDetails = groupsWithMemberCount.map((group) => {
-      const groupCreator = groupCreatorsDetails[group.createdBy];
-      return {
-        ...group,
-        firstName: groupCreator.first_name,
-        lastName: groupCreator.last_name,
-        image: groupCreator.picture?.url,
-      };
-    });
-
+    const dev = req.query.dev === "true";
+    if (dev) {
+      // Placing the new changes under the feature flag.
+      const discordId = req.userData?.discordId;
+      const groupsWithMembershipInfo = await discordRolesModel.enrichGroupDataWithMembershipInfo(discordId, groups);
```

Comment:
will you please give better name

Reply:
Can you suggest some? 
Generally, the groups do not have the data we need e.g. creator's firstName, lastName, image, memberCount, isMember (if the current user is a member of the group or not)

We are enriching existing group data with all the membership info, hence the name `enrichGroupDataWithMembershipInfo` and the variable name `groupsWithMembershipInfo `. In my opinion, it's apt for this.

I would love to hear your suggestion and thoughts on this.

Thank you for reviewing